content/guides/typescript-for-solid.mdx

```diff
@@ -0,0 +1,586 @@
+    "moduleResolution": "node",
+    "allowSyntheticDefaultImports": true,
+    "esModuleInterop": true,
+    "jsx": "preserve",
+    "jsxImportSource": "solid-js",
+    "types": ["vite/client"],
+    "noEmit": true,
+    "isolatedModules": true
+  }
+}
+```
+
+4. Create a TypeScript or `.tsx` file to test the setup.
+
+```typescript
+import { Component } from "solid-js";
+
+function MyTsComponent(): Component {
+  return (
+    <div>
+      <h1>This is a TypeScript component</h1>
+    </div>
+  );
+}
+
+export default MyTsComponent;
+```
+
+To use in an existing JavaScript component, import the TypeScript component:
+
+```javascript
+import MyTsComponent from "./MyTsComponent";
+
+function MyJsComponent() {
+  return (
+    <>
+      {/* ... */}
+      <MyTsComponent />
+    </>
+  );
+}
+```
+
+<Aside>
+**Note:** If you wish to change the entry point file from `index.jsx` to `index.tsx`, you need to modify the `src` attribute in `<script>` to look like the following:
+
+```htmlembedded
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <!-- ... -->
+  </head>
+  <body>
+    <noscript>You need to enable JavaScript to run this app.</noscript>
+    <div id="root"></div>
+    <script src="/src/index.tsx" type="module"></script>
+  </body>
+</html>
+```
+</Aside>
+
+## API types
+
+Solid is written in TypeScript, meaning everything is typed out of the box.
+
+The [API Documentation](/references) details the types of all the API calls. 
+In addition, there are several helpful type definitions to make it easier when you need to specify explicit types.
+
+### Signals
+
+Using `createSignal<T>`, you can define the signal's type as `T`. 
+
+```typescript
+const [count, setCount] = createSignal<number>();
+```
+
+Here, `createSignal` has the return type `Signal<number>`, corresponding to the type passed into it.
+This resolves to a getter-setter [tuple](https://www.typescriptlang.org/docs/handbook/2/objects.html#tuple-types), both of which are generically typed:
+
+```typescript
+import type { Signal, Accessor, Setter } from "solid-js";
+type Signal<T> = [get: Accessor<T>, set: Setter<T>];
+```
+
+In this instance, the signal getter, `count`, assumes a type `Accessor<number | undefined>`. 
+This denotes Solid's type `Accessor<T>`, mapping to `() => number | undefined`. 
+The `| undefined` is attributed to the lack of efault value in `createSignal`, leading to the signal's initial undefined state.
+
+The signal setter, `setCount`, is of type `Setter<number>`. 
+This complex type roughly resembles `(value? : number | ((prev?: number) => number)) => number`, accommodating both a direct number or a function which takes a potential pervious value that results in a number.
+Since the signal's start is undefined, both the direct number and the function's number parameter are optional.
+
+Typescript errors can present with parameter assignments, such as `setcount(value)`. 
+To make sure `value` isn't mistaken for a function, you can use `setCount(() => value)`. 
```

Comment:
Reading through this again, I don't like my suggested change either, since it doesn't really contextualize when this might happen -- with signals that store functions, and generic signals that could be functions. Without that context this is confusing.

Reply:
How do you suspect addomg that, then?

Reply:
```suggestion
When a signal can hold functions, its setter will not accept functions directly -- the Setter type disallows this -- 
as it has no way of knowing whether such functions should or should not be called to obtain the actual value. 
In such cases, you should always use the setter's callback form: `setSignal(() => value)`. 
```